components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java

```diff
@@ -599,6 +600,8 @@ public void run() {
                 MLModel model = modelBuilder.build();
                 
                 persistModel(id, ctxt.getModel().getName(), model);
+
+                emailTemplateParameters[1] = MLUtils.getLink(ctxt, MLConstants.MODEL_STATUS_COMPLETE);
                 EmailNotificationSender.sendModelBuildingCompleteNotification(emailNotificationEndpoint,
                         emailTemplateParameters);
             } catch (Exception e) {
```

Comment:
Its not a good idea to catch generic Exception. Its better to catch and handle the specific exception unless of course a third party API you depend on throws this Exception. In which case its better to comment and explain.